crates/http/Cargo.toml

```diff
@@ -10,128 +10,71 @@ license = "MIT OR Apache-2.0"
+[lints.rust]
+unexpected_cfgs = { level = "warn", check-cfg = ['cfg(coverage_nightly)'] }
 
+[dependencies]
+tokio = { version = "1.43.0" }
+thiserror = "2.0.11"
+futures = { version = "0.3.31", default-features = false }
+bon = "3.3.2"
+pin-project-lite = "0.2.16"
 scuffle-context.workspace = true
-scuffle-workspace-hack.workspace = true
 
-[features]
-error-backtrace = []
+# HTTP parsing
+http = "1.2.0"
+http-body = "1.0.1"
+bytes = "1.9.0"
 
-http1 = [
-    "hyper/http1",
-    "hyper-util/http1",
-]
+# Tracing
+tracing = { version = "0.1.41", optional = true }
 
-http2 = [
-    "hyper/http2",
-    "hyper-util/http2",
-]
-
-http3 = [
-    "h3",
-    "quic-quinn",
-]
+# HTTP/1.1, HTTP/2
+hyper = { version = "1.6.0", default-features = false, features = ["server"], optional = true }
+hyper-util = { version = "0.1.10", default-features = false, features = ["server", "tokio"], optional = true }
 
-http3-webtransport = [
-    "http3",
-    "quic-quinn",
-    "scuffle-h3-webtransport",
-]
+# QUIC + HTTP/3
+quinn = { version = "0.11.6", default-features = false, features = ["platform-verifier", "runtime-tokio", "rustls-ring", "aws-lc-rs"], optional = true }
+h3-quinn = { version = "0.0.7", default-features = false, optional = true }
+h3 = { version = "0.0.6", default-features = false, optional = true }
 
-# Quinn is a quic backend for http3
-quic-quinn = [
-    "http3",
-    "h3-quinn",
-    "quinn",
-]
+# TLS
+tokio-rustls = { version = "0.26.1", default-features = false, features = ["aws_lc_rs", "tls12"], optional = true }
+rustls = { version = "0.23.21", default-features = false, features = ["aws_lc_rs", "std", "tls12"], optional = true }
 
-tls-rustls = [
-    "rustls",
-    "tokio-rustls",
-]
+# Tower Services
+tower = { version = "0.5.2", default-features = false, features = ["make"], optional = true }
 
-tls-rustls-pem = [
-    "tls-rustls",
-    "rustls-pemfile",
-]
-
-tracing = [
-    "dep:tracing",
-]
-
-tower = [
-    "dep:tower-service",
-]
-
-axum = [
-    "dep:axum-core",
-    "tower",
-]
+scuffle-workspace-hack.workspace = true
 
-http3-default = [
-    "http3",
-    "quic-quinn",
-    "tls-rustls",
-]
+[dev-dependencies]
+# The http3 feature requires a custom build flag (see .cargo/config.toml)
+reqwest = { version = "0.12.12", default-features = false, features = ["rustls-tls", "http2", "http3"] }
+axum = { version = "0.8.1" }
+rustls-pemfile = "2.2.0"
+tokio-test = "0.4.4"
+scuffle-future-ext.workspace = true
 
-default = [
-    "http1",
-    "http2",
-    "tracing",
-]
+# https://doc.rust-lang.org/cargo/reference/features.html#dependency-features
+[features]
+default = ["http1", "http2", "tower"]
+tracing = ["dep:tracing", "quinn?/log", "h3-quinn?/tracing", "h3?/tracing", "tokio-rustls?/logging", "rustls?/logging"]
+http1 = ["dep:hyper", "dep:hyper-util", "hyper/http1", "hyper-util/http1"]
+http2 = ["dep:hyper", "dep:hyper-util", "hyper/http2", "hyper-util/http2"]
+http3 = ["dep:quinn", "dep:h3-quinn", "dep:h3", "tls-rustls"]
```

Comment:
I think this shouldn't add `tls-rustls` I think if someone use `http3` they should add both features manually. We should also add a `compile_error!` if `http3` is enabled without `tls-rustls` that way in the future we can have a alternative tls.